rwviewer/views/TextViewer.cpp

```diff
@@ -0,0 +1,251 @@
+#include "TextViewer.hpp"
+
+#include <render/TextRenderer.hpp>
+#include <rw/filesystem.hpp>
+
+#include <algorithm>
+#include <cctype>
+#include <iomanip>
+#include <iostream>
+
+#include <fonts/GameTexts.hpp>
+#include <loaders/LoaderGXT.hpp>
+#include <models/TextModel.hpp>
+#include <boost/filesystem.hpp>
+#include <render/GameRenderer.hpp>
+
+#include <QGroupBox>
+#include <QFormLayout>
+#include <QHBoxLayout>
+#include <QVBoxLayout>
+#include <QItemSelection>
+#include <QLineEdit>
+#include <QModelIndex>
+#include <QPushButton>
+#include <QRadioButton>
+#include <QRegExp>
+#include <QRegExpValidator>
+#include <QSpinBox>
+#include <QSplitter>
+#include <QTableView>
+#include <QTextEdit>
+
+void TextTableView::selectionChanged(const QItemSelection &selected, const QItemSelection &) {
+    if (!selected.size())
+        return;
+    auto index = selected.indexes()[0];
+    if (!index.isValid())
+        return;
+
+    auto *textModel = dynamic_cast<TextModel *>(this->model());
+    if (!textModel) {
+        return;
+    }
+    try {
+        auto gameString = textModel->lookupIndex(index);
+        emit gameStringChanged(gameString);
+    } catch (std::out_of_range &) {
+        return;
+    }
+}
+
+TextViewer::TextViewer(QWidget* parent, Qt::WindowFlags f)
+    : ViewerInterface(parent, f) {
+
+    auto dataLayout = new QVBoxLayout;
```

Comment:
Here and below I'd use value initialization.

Reply:
I know what you mean, but this is "the qt way".
All sample code of qt is written this way.